src/platform/windows/mod.rs

```diff
@@ -0,0 +1,1482 @@
+        let handle = dup_handle(*self.handle).unwrap();
+        OsIpcSharedMemory::from_handle(handle, self.length).unwrap()
+    }
+}
+
+impl PartialEq for OsIpcSharedMemory {
+    fn eq(&self, other: &OsIpcSharedMemory) -> bool {
+        unsafe {
+            self.length == other.length &&
+            memcmp(self.ptr as *const _, other.ptr as *const _, self.length) == 0
+        }
+    }
+}
+
+impl Deref for OsIpcSharedMemory {
+    type Target = [u8];
+
+    #[inline]
+    fn deref(&self) -> &[u8] {
+        assert!(!self.ptr.is_null() && self.handle.is_valid());
+        unsafe {
+            slice::from_raw_parts(self.ptr, self.length)
+        }
+    }
+}
+
+impl OsIpcSharedMemory {
+    #[allow(exceeding_bitshifts)]
+    fn new(length: usize) -> Result<OsIpcSharedMemory,WinError> {
+        unsafe {
+            let (lhigh, llow) = if cfg!(target_pointer_width = "64") {
+                ((length >> 32) as u32, (length & 0xffffffffusize) as u32)
+            } else {
+                (0, length as u32)
+            };
+            let handle =
+                kernel32::CreateFileMappingA(INVALID_HANDLE_VALUE,
+                                             ptr::null_mut(),
+                                             winapi::PAGE_READWRITE | winapi::SEC_COMMIT,
+                                             lhigh, llow,
+                                             ptr::null_mut());
+            if handle == INVALID_HANDLE_VALUE {
+                return Err(WinError::last("CreateFileMapping"));
+            }
+
+            OsIpcSharedMemory::from_handle(handle, length)
+        }
+    }
+
+    // There is no easy way to query the size of the mapping -- you
+    // can use NtQuerySection, but that's an undocumented NT kernel
+    // API.  Instead we'll just always pass the length along.
+    //
+    // This function takes ownership of the handle, and will close it
+    // when finished.
+    fn from_handle(handle_raw: HANDLE, length: usize) -> Result<OsIpcSharedMemory,WinError> {
+        // turn this into a WinHandle, because that will
+        // take care of closing it
+        let handle = WinHandle::new(handle_raw);
+        let address = unsafe {
+            kernel32::MapViewOfFile(handle_raw,
+                                    winapi::FILE_MAP_ALL_ACCESS,
+                                    0, 0, 0)
+        };
+        if address.is_null() {
+            return Err(WinError::last("MapViewOfFile"));
+        }
+
+        Ok(OsIpcSharedMemory {
+            handle: handle,
+            ptr: address as *mut u8,
+            length: length
+        })
+    }
+
+    pub fn from_byte(byte: u8, length: usize) -> OsIpcSharedMemory {
+        unsafe {
+            // panic if we can't create it
+            let mem = OsIpcSharedMemory::new(length).unwrap();
+            for element in slice::from_raw_parts_mut(mem.ptr, mem.length) {
+                *element = byte;
+            }
+            mem
```

Comment:
Why would you do this?  From my understanding, nothing prevents you from mutating the shared memory after you send it to another process; that even seems desirable.  If what we wanted was a way to one-way send big data, then we should explicitly bake that functionality in by allowing explicit buffers to be sent and let the implementation decide how to send them.  That would open up some optimization possibilities.